README.md

```diff
@@ -563,6 +565,44 @@ Brainstem provides a rich DSL for building presenters.  This section details the
     association :associated_objects, :polymorphic, "a mixture of objects related to this Widget"
   end
   ```
+  
+* `lookup` - Use this option to avoid N + 1 queries for Fields and Associations. The `lookup` lambda runs once when
+presenting and every presented model gets its assocation or value from the cache the `lookup` lambda generates. The
+`lookup` lambda takes in the presented models and should generate a cache containing the models' coresponding assocations
+or values. Brainstem expects the return result of the `lookup` to be a Hash where the keys are the presented models' ids
+and the values are those models' associations or values. Use the `lookup` when you would like to preload but cannot
+e.g. if your association references `current_user`. If the `lookup` opition is defined, the `dynamic` option is defined and
+one model is being presented, then the `dynamic` will be used. If multiple models are being presented and both options are defined,
+the `lookup` will be used.
+
+  ```ruby
+  associations do
+    association :current_user_groups, Group, "the Groups for the current user",
+                lookup: lambda { |models| 
+                  Group.where(subject_id: models.map(&:id)
+                    .where(user_id: current_user.id)
+                    .group_by { |group| group.subject_id } 
+                }
+  end
+	```
+* `lookup_fetch` - Use this option for Fields and Associations if you would like to override how a model should retrieve
+ its value or assocation from the cache returned by the `lookup` cache. The `lookup_fetch` lambda takes in the return
```

Comment:
The `lookup_fetch` lambda takes in the presented model and result from the `lookup` lambda